### ISSUE TITLE
Unable to run docker image on local GPU server with RTX 2080ti

### ISSUE THREAD

**Issue by MaxYousif** [huggingface/text-generation-inference#808]
### System Info

I am new to this framework of deploying LLMs, so please bare with and excuse my lack of expertise. If you happen to have useful resources to tackle the challenges I am experiencing then that would be much appreciated.

System Info
OS Version: 
```
(base) max@maxs-gpu-server:~/code/text-generation-inference$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 23.04
Release:	23.04
Codename:	lunar
```

Model being used: I've tried multiple, but the first one I attempted was `tiiuae/falcon-7b-instruct` (same as model in readme docs). I also attempted to use `TheBloke/Llama-2-7b-chat-fp16` but I think that model is too large for my GPU.

Hardware used: 1 local RTX 2080 ti GPU (11GB). AMD Threadripper 2920X CPU, with 64GB of RAM attached. 2TB SSD installed alongside. GPU details below: 
```
(base) max@maxs-gpu-server:~/code/text-generation-inference$ nvidia-smi
Thu Aug 10 12:16:55 2023       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.86.05              Driver Version: 535.86.05    CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 2080 Ti     Off | 00000000:42:00.0  On |                  N/A |
| 27%   32C    P8               2W / 250W |    427MiB / 11264MiB |      2%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      1804      G   /usr/lib/xorg/Xorg                          187MiB |
|    0   N/A  N/A      1983      G   /usr/bin/gnome-shell                         67MiB |
|    0   N/A  N/A      3420      G   ...5615870,15068953908461052321,262144      136MiB |
|    0   N/A  N/A      6836      G   ...ures=SpareRendererForSitePerProcess       22MiB |
|    0   N/A  N/A     13014      G   /usr/bin/nautilus                            10MiB |
+---------------------------------------------------------------------------------------+
```

Deployment specificities: Running via docker on my local GPU server. Docker version below
```
(base) max@maxs-gpu-server:~/code/text-generation-inference$ docker version
Client: Docker Engine - Community
 Cloud integration: v1.0.35-desktop+001
 Version:           24.0.5
 API version:       1.43
 Go version:        go1.20.6
 Git commit:        ced0996
 Built:             Fri Jul 21 20:35:14 2023
 OS/Arch:           linux/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.22.0 (117440)
 Engine:
  Version:          24.0.5
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.6
  Git commit:       a61e2b4
  Built:            Fri Jul 21 20:35:45 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Problem outline:
I am trying to run the `text-generation-inference` docker container on my own GPU server. Whenever I pass the `--gpu` argument, I get the following error:

```
(base) max@maxs-gpu-server:~/code/text-generation-inference$ docker run --gpus 0 -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:1.0.0 --model-id $model
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'legacy'
nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown.
ERRO[0000] error waiting for container: 
```
If I instead run the following without utilising the GPU, then the terminal hangs and the model weights are never downloaded.
```
(base) max@maxs-gpu-server:~/code/text-generation-inference$ docker run -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:1.0.0 --model-id $model
2023-08-10T09:05:53.912393Z  INFO text_generation_launcher: Args { model_id: "tiiuae/falcon-7b-instruct", revision: None, validation_workers: 2, sharded: None, num_shard: None, quantize: None, dtype: None, trust_remote_code: false, max_concurrent_requests: 128, max_best_of: 2, max_stop_sequences: 4, max_input_length: 1024, max_total_tokens: 2048, waiting_served_ratio: 1.2, max_batch_prefill_tokens: 4096, max_batch_total_tokens: None, max_waiting_tokens: 20, hostname: "e83a6ae9d25e", port: 80, shard_uds_path: "/tmp/text-generation-server", master_addr: "localhost", master_port: 29500, huggingface_hub_cache: Some("/data"), weights_cache_override: None, disable_custom_kernels: false, cuda_memory_fraction: 1.0, json_output: false, otlp_endpoint: None, cors_allow_origin: [], watermark_gamma: None, watermark_delta: None, ngrok: false, ngrok_authtoken: None, ngrok_edge: None, env: false }
2023-08-10T09:05:53.912621Z  INFO download: text_generation_launcher: Starting download process.
2023-08-10T09:05:57.446201Z  WARN text_generation_launcher: No safetensors weights found for model tiiuae/falcon-7b-instruct at revision None. Downloading PyTorch weights.

2023-08-10T09:05:57.753749Z  INFO text_generation_launcher: Download file: pytorch_model-00001-of-00002.bin

2023-08-10T09:10:17.088505Z ERROR text_generation_launcher: An error occurred while downloading using `hf_transfer`. Consider disabling HF_HUB_ENABLE_HF_TRANSFER for better error handling.

2023-08-10T09:10:17.088556Z  INFO text_generation_launcher: Retrying in 5 seconds

2023-08-10T09:10:22.093998Z  INFO text_generation_launcher: Retry 1/4

2023-08-10T09:10:22.094164Z  INFO text_generation_launcher: Download file: pytorch_model-00001-of-00002.bin
```

I appreciate that the following is difficult to reproduce as it is my own local set up. However, any insight into the above errors would be much appreciated. 

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

Steps to reproduce behaviour:
1. Simply run `docker run --gpus 0 -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:1.0.0 --model-id $model` on my local machine (so it will be difficult to reproduce)
2. To reproduce the hanging during model download, run `docker run -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:1.0.0 --model-id $model`

### Expected behavior

- Have the series of inference endpoints exposed on my local machine, including the `/docs` endpoint
- Inference utilises the GPU
- Be able to test the deployment with `curl`
- Download the model data into the attached `data` volume without hanging

**Comment by Narsil** [comment]
> nvidia-container-cli: initialization error: load library failed: libnvidia-ml.so.1: cannot open shared object file: no such file or directory: unknown.

This means you didn't install nvidia-docker properly, or your host cuda install doesn't work.

**Comment by Narsil** [comment]
The docker hanging is just because it's downloading the model and your internet connecction might be slow.

use `-e HF_HUB_ENABLE_HF_TRANSFER=0` to disable our DL optimizations to see the actual model being downloaded

**Comment by maxyousif15** [comment]
Thanks @Narsil. This issue can be closed because I have managed to solve the problems I've outlined; I now have the TGI container running on my local GPU server and I have managed to run inference successfully! For anyone else who is reading this, the following outlines the solutions:

1. Downgrade from Ubuntu 23.04 to 22.04. Firstly, 23.04 was very glitchy for me. Additionally, I don't think nvidia-docker is supported for Ubuntu 23.04 distribution. Well, at least it is not outlined [here](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html). I think all other requirements (nvidia-driver, cuda etc) may be supported on Ubuntu 23.04 but I couldn't get it to download & install nvidia-docker for 23.04 distribution. 
2. I did not install Docker Desktop after downgrading to 22.04. Instead, I installed it following this [approach](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository) (install via apt repository). I am not sure if this had something to do with my failures, but Docker Desktop was not mentioned in any of the tutorials/blogs I followed. 

The following [gist](https://gist.github.com/amir-saniyan/b3d8e06145a8569c0d0e030af6d60bea) is an invaluable resource to follow to get set up. 

Regarding docker hanging - I will trial that and see. My internet is very fast (gigabit speed) so that wouldn't be the cause. Hopefully the logs will inform me of what I need to do in case of errors. Thanks again for your help, much appreciated.
